File: PHP/RESTapi-MySQL/rest.php
<?php
include_once 'database.php';

//Create database connection
$database = new Database();
$conn = $database->getConnection();

//get the in-data
$data = json_decode(file_get_contents("php://input"));

//check what type of request
switch($_SERVER['REQUEST_METHOD'])
{
//If there is a get request
case 'GET': $the_request = &$_GET;
    //If there is no inparameters
    if(empty($_GET)){
        //Get all flightdatafrom the darabase
        $query = "SELECT * FROM flightdata";
        $stmt = $conn->query($query);
       
        //adding flightdata to a array
        $flightdata = array();
        while ($row = $stmt->fetch()) {
            $tmp = array("id" => $row['id']);
            $tmp += ["airline" => $row['airline']];
            $tmp += ["airlineId" => $row['airlineId']];
            $tmp += ["sourceAirport" => $row['sourceAirport']];
            $tmp += ["sourceAirportId" => $row['sourceAirportId']];
            $tmp += ["destinationAirport" => $row['destinationAirport']];
            $tmp += ["destinationAirportId" => $row['destinationAirportId']];
            $tmp += ["stops" => $row['stops']];
            $tmp += ["equipment" => $row['equipment']];
            array_push($flightdata, $tmp);
        }
       
        //responed with the array with all flightdata
        http_response_code(200);
        echo json_encode($flightdata, JSON_PRETTY_PRINT);
    
    //If there is inparameters
    }else{
        //Get the inparameters into an array
        $get_arr = array(
            "id" =>  isset($_GET['id']) ? $_GET['id'] : null,
            "airline" => isset($_GET['airline']) ? $_GET['airline'] : null,
            "airlineId" => isset($_GET['airlineId']) ? $_GET['airlineId'] : null,
            "sourceAirport" => isset($_GET['sourceAirport']) ? $_GET['sourceAirport'] : null,
            "sourceAirportId" => isset($_GET['sourceAirportId']) ? $_GET['sourceAirportId'] : null,
            "destinationAirport" => isset($_GET['destinationAirport']) ? $_GET['destinationAirport'] : null,
            "destinationAirportId" => isset($_GET['destinationAirportId']) ? $_GET['destinationAirportId'] : null,
            "stops" => isset($_GET['stops']) ? $_GET['stops'] : null,
            "equipment" => isset($_GET['equipment']) ? $_GET['equipment'] : null
    
        );
        
        //get the matching fligtdata from the database
        $query = "SELECT * FROM flightdata WHERE id LIKE :id OR airline LIKE :airline OR airlineId LIKE :airlineId OR sourceAirport LIKE :sourceAirport OR sourceAirportId LIKE :sourceAirportId OR destinationAirport LIKE :destinationAirport OR destinationAirportId LIKE :destinationAirportId OR stops LIKE :stops OR equipment LIKE :equipment";
        $stmt = $conn->prepare($query);
        $stmt->execute(["id" => $get_arr['id'], "airline" => $get_arr['airline'], "airlineId" => $get_arr['airlineId'], "sourceAirport" => $get_arr['sourceAirport'], "sourceAirportId" => $get_arr['sourceAirportId'], "destinationAirport" => $get_arr['destinationAirport'], "destinationAirportId" => $get_arr['destinationAirportId'], "stops" => $get_arr['stops'], "equipment"=> $get_arr['equipment']]); 
        
        //adding flightdata to a array
        $flightdata = array();
        while ($row = $stmt->fetch()) {
            $tmp = array("id" => $row['id']);
            $tmp += ["airline" => $row['airline']];
            $tmp += ["airlineId" => $row['airlineId']];
            $tmp += ["sourceAirport" => $row['sourceAirport']];
            $tmp += ["sourceAirportId" => $row['sourceAirportId']];
            $tmp += ["destinationAirport" => $row['destinationAirport']];
            $tmp += ["destinationAirportId" => $row['destinationAirportId']];
            $tmp += ["stops" => $row['stops']];
            $tmp += ["equipment" => $row['equipment']];
            array_push($flightdata, $tmp);
        }

        //If there is matching flightdata in the database
        if(!empty($flightdata)){
            http_response_code(200);
            //responed with the array with the flightdata
            echo json_encode($flightdata, JSON_PRETTY_PRINT);
        //If there is no matching flightdata in the database
        }else{
            http_response_code(404);
            echo json_encode(array("message" => "No flightdata matched the get request"), JSON_PRETTY_PRINT);
        }
    }
    break;

//If there is a post request
case 'POST': $the_request = &$_POST;
    //Only works if there is data
    if(!empty($data)){
        //Preparing the sql statement with the in-data
        $query = "INSERT INTO flightdata SET airline=:airline, airlineId=:airlineId, sourceAirport=:sourceAirport, sourceAirportId=:sourceAirportId, destinationAirport=:destinationAirport, destinationAirportId=:destinationAirportId, stops=:stops, equipment=:equipment";
        $stmt = $conn->prepare($query);
        $stmt->bindParam(":airline", $data->airline);
        $stmt->bindParam(":airlineId", $data->airlineId);
        $stmt->bindParam(":sourceAirport", $data->sourceAirport);
        $stmt->bindParam(":sourceAirportId", $data->sourceAirportId);
        $stmt->bindParam(":destinationAirport", $data->destinationAirport);
        $stmt->bindParam(":destinationAirportId", $data->destinationAirportId);
        $stmt->bindParam(":stops", $data->stops);
        $stmt->bindParam(":equipment", $data->equipment);

        //Executes the sql statement and checks for errors
        if($stmt->execute()){
            http_response_code(200);
            echo json_encode(array("message" => "flightdata was created."));
        }else{
            http_response_code(503);
            echo json_encode(array("message" => "Unable to create flightdata."));
        }
    }else{
        //Respones if there is no in-data
        http_response_code(204);
        echo json_encode(array("message" => "Unable to create flightdata. Data is incomplete."));
    }
    break;

 // Etc.
default:
    http_response_code(503);
}
?>

File: Python-Flask/RESTapi-MySQL/rest.py
from flask import Flask,request, jsonify
import mysql.connector
import json

app = Flask(__name__)

#Create database connection
db = mysql.connector.connect(
  host="localhost",
  user="root",
  password="",
  database="examensarbete"
)

#If there is a get request
@app.route('/', methods=['GET'])
def get():
    #If there is inparameters
    if request.args:
        #Get the inparameters into an dictionary
        tmp = {}
        tmp["id"]  = request.args.get('id')
        tmp["airline"]  = request.args.get('airline')
        tmp["airlineId"]  = request.args.get('airlineId')
        tmp["sourceAirport"]  = request.args.get('sourceAirport')
        tmp["sourceAirportId"]  = request.args.get('sourceAirportId')
        tmp["destinationAirport"]  = request.args.get('destinationAirport')
        tmp["destinationAirportId"]  = request.args.get('destinationAirportId')
        tmp["stops"]  = request.args.get('stops')
        tmp["equipment"]  = request.args.get('equipment')

        #Get the matching fligtdata from the database
        cursor = db.cursor()
        cursor.execute(f"SELECT * FROM flightdata WHERE id LIKE '{tmp.get('id')}' OR airline LIKE '{tmp.get('airline')}' OR airlineId LIKE '{tmp.get('airlineId')}' OR sourceAirport LIKE '{tmp.get('sourceAirport')}' OR sourceAirportId LIKE '{tmp.get('sourceAirportId')}' OR destinationAirport LIKE '{tmp.get('destinationAirport')}' OR destinationAirportId LIKE '{tmp.get('destinationAirportId')}' OR stops LIKE '{tmp.get('stops')}' OR equipment LIKE '{tmp.get('equipment')}'")
        result = cursor.fetchall()

        #adding flightdata to a list
        flightdata = []
        for row in result:
            tmp = {}
            tmp["id"] = row[0]
            tmp["airline"] = row[1]
            tmp["airlineId"] = row[2]
            tmp["sourceAirport"] = row[3]
            tmp["sourceAirportId"] = row[4]
            tmp["destinationAirport"] = row[5]
            tmp["destinationAirportId"] = row[6]
            tmp["stops"] = row[7]
            tmp["equipment"] = row[8]
            flightdata.append(tmp)

        #If there is matching flightdata in the database
        if flightdata:
            #responed with the list with the flightdata
            return jsonify(flightdata),200
        #If there is no matching flightdata in the database
        else:
            return jsonify({"message": "No flightdata matched the get request"}),404
    
    #If there is no inparameters
    else:
        #Get all flightdatafrom the darabase
        cursor = db.cursor()
        cursor.execute("SELECT * FROM flightdata")
        result = cursor.fetchall()

        #Adding flightdata to a list
        flightdata = []
        for row in result:
            tmp = {}
            tmp["id"] = row[0]
            tmp["airline"] = row[1]
            tmp["airlineId"] = row[2]
            tmp["sourceAirport"] = row[3]
            tmp["sourceAirportId"] = row[4]
            tmp["destinationAirport"] = row[5]
            tmp["destinationAirportId"] = row[6]
            tmp["stops"] = row[7]
            tmp["equipment"] = row[8]
            flightdata.append(tmp)

        #Responed with the list with all flightdata
        return jsonify(flightdata),200

#If there is a post request
@app.route('/', methods=['POST'])
def post():

    #Only works if there is in-data
    if request.data:

        #converts in-data to a dictionary 
        data = json.loads(request.data)

        #if there is multiple flightdata objects in the data
        if type(data) == list:
            #loops throut the flightdata
            for obj in data:
                #Preparing the sql statement with the in-data
                cursor = db.cursor()
                query = "INSERT INTO flightdata SET airline=%s, airlineId=%s, sourceAirport=%s, sourceAirportId=%s, destinationAirport=%s, destinationAirportId=%s, stops=%s, equipment=%s";
                val = (obj["airline"], obj["airlineId"], obj["sourceAirport"], obj["sourceAirportId"], obj["destinationAirport"], obj["destinationAirportId"], obj["stops"], obj["equipment"])
                cursor.execute(query, val)

                #Executes the sql statement and checks for errors
                try:
                    db.commit()

                except mysql.connector.Error as error :
                    return jsonify({"message": "Unable to create flightdata."}),503
                
            return jsonify({"message": "flightdata was created."}),200
        
        #if there is one flightdata object in the data
        else:
            #Preparing the sql statement with the in-data
            cursor = db.cursor()
            query = "INSERT INTO flightdata SET airline=%s, airlineId=%s, sourceAirport=%s, sourceAirportId=%s, destinationAirport=%s, destinationAirportId=%s, stops=%s, equipment=%s";
            val = (data["airline"], data["airlineId"], data["sourceAirport"], data["sourceAirportId"], data["destinationAirport"], data["destinationAirportId"], data["stops"], data["equipment"])
            cursor.execute(query, val)
        
            #Executes the sql statement and checks for errors
            try:
                db.commit()
                return jsonify({"message": "flightdata was created."}),200

            except mysql.connector.Error as error :
                return jsonify({"message": "Unable to create flightdata."}),503

    #Respones if there is no in-data
    else:
        return jsonify({"message": "Unable to create flightdata. Data is incomplete."}),204
